### PR TITLE
Static generator now implemented in jython

### PR DESCRIPTION
**Pull request by mpdickie**

### CODE REVIEW

File: org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java [hunk 1]
```diff
@@ -65,6 +65,10 @@ public void run() {
         return new JythonObjectFactory<>(ScanPointIterator.class, "jython_spg_interface", "JArrayGenerator");
     }
 
+	public static JythonObjectFactory<ScanPointIterator> JStaticGeneratorFactory() {
```
Comment (gerring) [inline org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

File: org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java [hunk 2]
```diff
@@ -65,6 +65,10 @@ public void run() {
...
+		return new JythonObjectFactory<>(ScanPointIterator.class, "jython_spg_interface", "JStaticPointGenerator");
+	}
+
 	public static JythonObjectFactory<ScanPointIterator> JSpiralGeneratorFactory() {
```
Comment (gerring) [inline org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

**Comment by gerring** [comment]
SonarQube analysis reported 17 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 15 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ScanPointGeneratorFactory.java#L57](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L57): Define a constant instead of duplicating this literal "jython_spg_interface" 15 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [ScanPointGeneratorFactory.java#L23](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L23): Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L56](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L56): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L60](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L60): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L64](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L64): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L76](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L76): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L80](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L80): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L84](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L84): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L88](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L88): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L92](https://github.com/eclipse/scanning/blob/4213237594d0951fbed6d63baaeb9ca16b072946/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L92): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by gerring** [comment]
SonarQube analysis reported 17 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 15 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ScanPointGeneratorFactory.java#L57](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L57): Define a constant instead of duplicating this literal "jython_spg_interface" 15 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [ScanPointGeneratorFactory.java#L23](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L23): Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L56](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L56): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L60](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L60): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L64](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L64): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L76](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L76): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L80](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L80): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L84](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L84): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L88](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L88): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L92](https://github.com/eclipse/scanning/blob/4d0180a07a74ed37523047a7fc51c96c88ae49a0/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L92): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java [hunk 3]
```diff
@@ -81,7 +80,7 @@ private void checkNexusFile(IRunnableDevice<ScanModel> scanner) throws Exception
 		NXentry entry = rootNode.getEntry();
 		NXinstrument instrument = entry.getInstrument();
 
-		assertSolsticeScanGroup(entry, false, false);
+//		assertSolsticeScanGroup(entry, false, false, 1);
```
Comment (gerring) [inline org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/StaticPosition.java [hunk 4]
```diff
@@ -46,13 +46,20 @@ public int getIndex(String name) {
...
 	}
 
 	@Override
 	public int getScanRank() {
+		return 1;
+	}
+
+	@Override
+	public int getIndex(int dimension) {
+		// TODO: how do we get the index when the number of StaticPositions is > 1?
```
Comment (gerring) [inline org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/StaticPosition.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

**Comment by gerring** [comment]
SonarQube analysis reported 25 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 15 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ScanPointGeneratorFactory.java#L57](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L57): Define a constant instead of duplicating this literal "jython_spg_interface" 15 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [MandelbrotAcquireTest.java#L63](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L63): Make the enclosing method "static" or remove this set. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)
1. ![MAJOR][MAJOR] [ScanPointGeneratorFactory.java#L23](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L23): Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L61](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L61): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L92](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L92): Rename "detector" which hides the field declared at line 58. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L139](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L139): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L151](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L151): Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L159](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L159): Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L56](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L56): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L60](https://github.com/eclipse/scanning/blob/24840d328af4889123d8dedb88e22cf47e3d2de9/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L60): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Mark-Booth** [comment]
With the target platform fix in my pull request having been merged, this is now building again.

Tests are failing however:

> Tests run: 94, Failures: 2, Errors: 0, Skipped: 3, Time elapsed: 56.763 sec <<< FAILURE! - in org.eclipse.scanning.test.scan.nexus.Suite

**Comment by mpdickie** [comment]
Tests are currently failing for this change as the new jython static generator's getRank and getShape methods return 0 and 'new int[] { size }' respectively, even when size is 1. The previous Java version returned 0 and an empty int array respectively from these two methods. I'm not sure which is correct, but I think the second version, but I couldn't convince anyone else of this. Therefore this change will stay parked here for the moment as fixing this requires a bit of work.

**Comment by Mark-Booth** [comment]
No problem @mpdickie , I just wanted to let you know that the previous build problems had been dealt with.

**Comment by gerring** [comment]
SonarQube analysis reported 25 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 15 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ScanPointGeneratorFactory.java#L57](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L57): Define a constant instead of duplicating this literal "jython_spg_interface" 15 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [MandelbrotAcquireTest.java#L63](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L63): Make the enclosing method "static" or remove this set. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)
1. ![MAJOR][MAJOR] [ScanPointGeneratorFactory.java#L23](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L23): Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L61](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L61): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L92](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L92): Rename "detector" which hides the field declared at line 58. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L139](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L139): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L151](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L151): Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR][MAJOR] [MandelbrotAcquireTest.java#L159](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/MandelbrotAcquireTest.java#L159): Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L56](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L56): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [ScanPointGeneratorFactory.java#L60](https://github.com/eclipse/scanning/blob/f55640adfce9471eda70de7d2894b9f87e12a68f/org.eclipse.scanning.points/src/org/eclipse/scanning/points/ScanPointGeneratorFactory.java#L60): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**